src/components/MainNav/MyProfile/MyProfile.js

```diff
@@ -47,6 +55,32 @@ export default class MyProfile extends Component {
     this.getProfileImage = this.getProfileImage.bind(this);
   }
 
+  componentDidMount() {
+    const links = this.props.modules.settings.map(setting => setting.links);
+
+    if (links && Array.isArray(links)) {
+      const linkItems = links[0].userDropdown;
+
+      linkItems.forEach((item, index) => {
+        this.isShownToUser().then((response) => {
+          const showToUser = response.status === 200;
+
+          if (showToUser) {
+            const itemCaption = item.caption;
+            const buttonId = 'clickable-chagepw' + index;
```

Comment:
Hardwiring `changepw` into the ID can't be right -- this is a general-purpose facility for adding links into the user dropdown menu, and it merely _happens_ that we are using it at the moment to implement a change-password link. Something like `'clickable-usermenu' + index` would be better.

Reply:
changed